pkg/build/api/v1/types.go

```diff
@@ -402,6 +408,17 @@ type SourceBuildStrategy struct {
 	ForcePull bool `json:"forcePull,omitempty"`
 }
 
+// JenkinsPipelineBuildStrategy holds parameters specific to a Jenkins Pipeline build.
+type JenkinsPipelineBuildStrategy struct {
```

Comment:
Naming wise, unless someone has a counter example / argument, this should be "JenkinsfileBuildStrategy" and the corresponding changes should trickle through.  We commit to supporting "Jenkinsfiles" in all their glory with this strategy.